File: algorithms/sorting/tutorial-intro/tutorial-intro-javascript/js/testeIntroTutorial.js


import {introTutorial} from '../js/introTutorial.js';
import assert from 'assert';

describe('introTutorial', function() {

    it('O indice de V em arr é: 2', function() {

        let V = 3;
        let arr = [1, 2, 3];
        
        let esperado = 2;
        
        let resultado = introTutorial(V, arr);

        assert.strictEqual(esperado, resultado);
    });
});

describe('introTutorial', function() {

    it('O indice de V em arr é: 1', function() {

        let V = 4;
        let arr = [1, 4, 5, 7, 9, 12];
        
        let esperado = 1;
        
        let resultado = introTutorial(V, arr);

        assert.strictEqual(esperado, resultado);
    });
});


describe('introTutorial', function() {

    it('O indice de V em arr é: 11', function() {

        let V = 23;
        let arr = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23];
        
        let esperado = 11;
        
        let resultado = introTutorial(V, arr);

        assert.strictEqual(esperado, resultado);
    });
});

File: algorithms/sorting/tutorial-intro/tutorial-intro-java/tutorial-intro-java/src/TesteIntroTutorial.java
import static org.junit.Assert.assertEquals;

import java.util.Arrays;
import java.util.List;

import org.junit.Test;

public class TesteIntroTutorial {

    @Test
    public void testeCaso1() {

        int V = 3;
        List<Integer> arr = Arrays.asList(1, 2, 3);
        
        int esperado = 2;
        
        int resultado = Resultado.introTutorial(V, arr);

        assertEquals(esperado, resultado);
    }
    
    @Test
    public void testeCaso2() {

        int V = 4;
        List<Integer> arr = Arrays.asList(1, 4, 5, 7, 9, 12);
        
        int esperado = 1;
        
        int resultado = Resultado.introTutorial(V, arr);

        assertEquals(esperado, resultado);
    }

    @Test
    public void testeCaso3() {

        int V = 23;
        List<Integer> arr = Arrays.asList(1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23);
        
        int esperado = 11;
        
        int resultado = Resultado.introTutorial(V, arr);

        assertEquals(esperado, resultado);
    }
}
